### PR TITLE
issue: 1057922 Optimize socket m_rx_ring_map_lock usage

### PR DESCRIPTION
**Pull request by liranoz12**
- Convert m_rx_ring_map_lock type to lock_mutex.
- TCP: lock only when iterate the ring map.

Signed-off-by: Liran Oz <lirano@mellanox.com>

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/1892/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/1331/ for details (Mellanox internal link).

**Comment by liranoz12** [comment]
bot:retest

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/1906/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/1344/ for details (Mellanox internal link).

**Comment by liranoz12** [comment]
@igor-ivanov ,
I will check this, thanks.

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/1450/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/1451/ for details (Mellanox internal link).

File: src/vma/sock/sockinfo_tcp.cpp [hunk 1]
```diff
@@ -3008,20 +3008,15 @@ bool sockinfo_tcp::is_readable(uint64_t *p_poll_sn, fd_array_t* p_fd_array)
...
+		if (m_ready_conn_cnt) {
 			si_tcp_logdbg("accept ready");
-			goto noblock_nolock;
+			return true;
 		}
 
-		if (m_sock_state == TCP_SOCK_ACCEPT_SHUT) goto noblock_nolock;
-
-		return false;
+		return m_sock_state == TCP_SOCK_ACCEPT_SHUT ? true : false;
```
Comment (igor-ivanov) [inline src/vma/sock/sockinfo_tcp.cpp]: could you consider wrapping with (...) to improve visibility (it is not mandatory)

File: src/vma/sock/sockinfo_tcp.cpp [hunk 2]
```diff
@@ -3030,29 +3025,33 @@ bool sockinfo_tcp::is_readable(uint64_t *p_poll_sn, fd_array_t* p_fd_array)
...
-		   if (ret <= 0) {
-			   break;
-		   }
+
+	if (likely(m_p_rx_ring)) {
+		while(!g_b_exit && is_rtr()) {
+			// likely scenario: rx socket bound to specific cq
+			ret = m_p_rx_ring->poll_and_process_element_rx(p_poll_sn, p_fd_array);
+			if (m_n_rx_pkt_ready_list_count)
+				return true;
```
Comment (igor-ivanov) [inline src/vma/sock/sockinfo_tcp.cpp]: please use `if () {}`

File: src/vma/sock/sockinfo_tcp.cpp [hunk 3]
```diff
@@ -3061,21 +3060,18 @@ bool sockinfo_tcp::is_readable(uint64_t *p_poll_sn, fd_array_t* p_fd_array)
...
 				//g_p_lwip->do_timers();
 				ret = p_ring->poll_and_process_element_rx(p_poll_sn, p_fd_array);
-				if (m_n_rx_pkt_ready_list_count)
-					goto noblock;
+				if (m_n_rx_pkt_ready_list_count) {
+					m_rx_ring_map_lock.unlock();
+					return true;
+				}
 				if (ret <= 0)
 					break;
```
Comment (igor-ivanov) [inline src/vma/sock/sockinfo_tcp.cpp]: please use `if () {}` to be consistent with line 3048

File: src/vma/sock/sockinfo_tcp.cpp [hunk 4]
```diff
@@ -3061,21 +3060,18 @@ bool sockinfo_tcp::is_readable(uint64_t *p_poll_sn, fd_array_t* p_fd_array)
...
-	if (!m_n_rx_pkt_ready_list_count) {
 		m_rx_ring_map_lock.unlock();
-		return false;
 	}
-noblock:
-	m_rx_ring_map_lock.unlock();
-noblock_nolock:
-	return true;
+
+	return m_n_rx_pkt_ready_list_count ? true : false;
```
Comment (igor-ivanov) [inline src/vma/sock/sockinfo_tcp.cpp]: could you consider wrapping with (...) to improve visibility (it is not mandatory)

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/1544/ for details (Mellanox internal link).

**Comment by liranoz12** [comment]
No performance improvement observed.

**Pull request closed without merging**